### PR TITLE
[SEMVER-MAJOR] add no-extra-semi

### PR DESCRIPTION
**Pull request by jannyHou**
### Description

A follow up PR based on comment https://github.com/strongloop/generator-loopback/pull/363#discussion_r213294275

### CODE REVIEW

File: eslint.json [hunk 1]
```diff
@@ -56,6 +56,7 @@
     "quotes": ["error", "single", "avoid-escape"],
     "semi-spacing": ["error", {"before": false, "after": true}],
     "semi": ["error", "always"],
+    "no-extra-semi": "error",
```
Comment (bajtos) [inline eslint.json]: I believe we are keeping the options sorted by alphabet, i.e. `no-extra-semi` should be inside the group of other `no-*` options.

**Review by bajtos — COMMENTED** [review]
Please note this is a breaking change that should be released in a semver-major release.

**Comment by jannyHou** [comment]
> Please note this is a breaking change that should be released in a semver-major release.

@bajtos I see. Yeah we should upgrade all LB3 modules by applying this new rule after releasing a major version.

**Review by bajtos — APPROVED** [review]
:shipit:

**Review by virkt25 — APPROVED** [review]

**Comment by rmg** [comment]
All of the downstream CI failures here indicate failures that will happen when this PR is merged. To properly reflect the semver-major nature of the change, `package.json` should have been modified to _actually_ bump the major version.

**Pull request merged**